src/lib/netlist/devices/nld_4029.cpp

```diff
@@ -0,0 +1,283 @@
+ *          |    6  ||  0 |  1 |  1 |  0 |  1 ||     7 |
+ *          |    7  ||  0 |  1 |  1 |  1 |  1 ||     8 |
+ *          |    8  ||  1 |  0 |  0 |  0 |  1 ||     9 |
+ *          |    9  ||  1 |  0 |  0 |  1 |0|CI||     0 |
+ *          |    A  ||  1 |  0 |  1 |  0 |  1 ||     B |
+ *          |    B  ||  1 |  0 |  1 |  1 |0|CI||     6 |
+ *          |    C  ||  1 |  1 |  0 |  0 |  1 ||     D |
+ *          |    D  ||  1 |  1 |  0 |  1 |0|CI||     4 |
+ *          |    E  ||  1 |  1 |  1 |  0 |  1 ||     F |
+ *          |    F  ||  1 |  1 |  1 |  1 |0|CI||     2 |
+ *          |    0  ||  0 |  0 |  0 |  0 |  1 ||     1 |
+ *          +-------++----+----+----+----+----++-------+
+ *
+ *          B/D low (Decimal), U/D low (Down)
+ *          +-------++----+----+----+----+----++-------+
+ *          | COUNT || Q4 | Q3 | Q2 | Q1 | CO ||  NEXT |
+ *          +=======++====+====+====+====+====++=======+
+ *          |    F  ||  1 |  1 |  1 |  1 |  1 ||     E |
+ *          |    E  ||  1 |  1 |  1 |  0 |  1 ||     D |
+ *          |    D  ||  1 |  1 |  0 |  1 |  1 ||     C |
+ *          |    C  ||  1 |  1 |  0 |  0 |  1 ||     B |
+ *          |    B  ||  1 |  0 |  1 |  1 |  1 ||     A |
+ *          |    A  ||  1 |  0 |  1 |  0 |  1 ||     9 |
+ *          |    9  ||  1 |  0 |  0 |  1 |  1 ||     8 |
+ *          |    8  ||  1 |  0 |  0 |  0 |  1 ||     7 |
+ *          |    7  ||  0 |  1 |  1 |  1 |  1 ||     6 |
+ *          |    6  ||  0 |  1 |  1 |  0 |  1 ||     5 |
+ *          |    5  ||  0 |  1 |  0 |  1 |  1 ||     4 |
+ *          |    4  ||  0 |  1 |  0 |  0 |  1 ||     3 |
+ *          |    3  ||  0 |  0 |  1 |  1 |  1 ||     2 |
+ *          |    2  ||  0 |  0 |  1 |  0 |  1 ||     1 |
+ *          |    1  ||  0 |  0 |  0 |  1 |  1 ||     0 |
+ *          |    0  ||  0 |  0 |  0 |  0 |0|CI||     9 |
+ *          |    9  ||  1 |  0 |  0 |  1 |  1 ||     8 |
+ *          +-------++----+----+----+----+----++-------+
+ *
+ *          Note that in all cases where Carry-out is generating a non-1 signal (0 | Carry-in),
+ *          this signal is asynchronous, so if carry-in changes the carry-out value will
+ *          change immediately.
+ *
+ *
+ *          Preset/Carry in function table
+ *
+ *          +-----+-----+-----++----+----+----+----+
+ *          | PE  | CLK | CI  || Q1 | Q2 | Q3 | Q4 |
+ *          +=====+=====+=====++====+====+====+====+
+ *          |  1  |  X  |  X  || J1 | J2 | J3 | J4 |
+ *          |  0  |  X  |  1  || Q1 | Q2 | Q3 | Q4 |
+ *          |  0  | ./` |  0  ||       COUNT       |
+ *          +-----+-----+-----++----+----+----+----+
+ *
+ *  Naming conventions follow National Semiconductor datasheet
+ *
+ */
+
+#include "nld_4029.h"
+#include "nl_base.h"
+
+namespace netlist
+{
+namespace devices
+{
+
+	// delay is just copied from nld_7492 and assumed to be similar for output propagation
+	static constexpr const std::array<netlist_time, 4> delay =
```

Comment:
CMOS device - if there would be a delay it would be an order slower. However I doubt there is delay. The 7492 consists of 4 chained flip-flops. This is an up-down counter and as long as the datasheet doesn't state something different I assume the outputs are synchronous.

Reply:
I changed this to a fixed delay of 320ns